posthog/temporal/workflows/s3_batch_export.py

```diff
@@ -119,10 +124,21 @@ def start(self) -> str:
         if self.is_upload_in_progress() is True:
             raise UploadAlreadyInProgressError(self.upload_id)
 
-        multipart_response = self.s3_client.create_multipart_upload(Bucket=self.bucket_name, Key=self.key)
-        self.upload_id = multipart_response["UploadId"]
+        optional_kwargs = {}
+        if self.encryption:
+            optional_kwargs["ServerSideEncryption"] = self.encryption
+        if self.kms_key_id:
+            optional_kwargs["SSEKMSKeyId"] = self.kms_key_id
 
-        return self.upload_id
+        multipart_response = self.s3_client.create_multipart_upload(
+            Bucket=self.bucket_name,
+            Key=self.key,
+            **optional_kwargs,
+        )
+        upload_id: str = multipart_response["UploadId"]
+        self.upload_id = upload_id
```

Comment:
why not
```suggestion
        self.upload_id = multipart_response["UploadId"]
```

Reply:
Type checking fails as `self.upload_id` is `str | None` and this function returns `str`, not `None`, so we need the extra `upload_id: str`

Reply:
shouldn't that work just fine - if it's str | None you can assign it str, I'm confused

Reply:
You can assign it `str` but then you cannot return it, so you need the extra variable. Not the only way to solve this though, you could have some `isinstance` checks, adding an extra variable seemed like the easiest